### ISSUE TITLE
Distinguish empty and nonempty lists

### ISSUE THREAD

**Issue by mcclure** [J-F-Liu/pom#70]
I forgot `list()` was in pom. So for a while I was using a construct like

`(node() + (blank() * node()).repeat(0..)).map(tuple_merge)`

Where `tuple_merge` was a function that took `(T, Vec<T>)` and prefixed `T` onto the vector. This is silly. I have now switched to `list(node(), blank())`

Problem!

When I was using my terrible `tuple_merge` construct, I could say *either* `0..` or `1..`. I could distinguish "0 or more" from "1 or more".

`list()`'s documentation says only "Parse separated list.". Is this a list of 0 or more, or a list of 1 or more? How do I force it to one or the other?

I suggest:

- Clarify in the documentation if `list()` is 0-or-more or 1-or-more
- Any one of the following feature requests:
    - Split `list()` into `list()` (1-or-more) or `opt_list()` (0-or-more)
    - Change `list()` to take a third parameter, a range, such as `0..`, `1..`, or `2..4`, like `repeat()` does.
    - Split `list` into `list()` (whatever list() is now) and `list_range(parser, separator, range)` where range is `3..` or whatever.

The second option, the range, would be I think better overall

**Comment by mcclure** [comment]
So in my testing:

- It appears list() is "zero-or-more"
- I was able to effectively create a one-or-more list with

```
fn require_nonzero<T>(v:Vec<T>) -> pom::Result<Vec<T>> {
    if v.len() > 1 {
        Ok(v)
    } else {
        Err(pom::Error::Incomplete)
    }
}

// later...
list(pattern1, pattern2).convert(require_nonzero)
```

Is this the recommended approach? Should something like this be documented?

**Comment by J-F-Liu** [comment]
Yes,  list() is "zero-or-more", for one-or-more list I prefer `item() + (sep() * item()).repeat(0..)`.
